src/main/java/com/example/demo/service/impl/PolygonServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.example.demo.service.impl;
+
+import com.example.demo.dto.PolygonDto;
+import com.example.demo.dto.PolygonWithAreaDto;
+import com.example.demo.entity.Point;
+import com.example.demo.entity.Polygon;
+import com.example.demo.repository.PolygonRepository;
+import com.example.demo.service.PolygonService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PolygonServiceImpl implements PolygonService {
+    private final PolygonRepository repository;
+    private final ObjectMapper mapper;
+
+    @Override
+    public boolean add(PolygonDto dto) {
+        List<Point> points = dto.getPoints();
+        if (points.get(0).equals(points.get(points.size() - 1))) {
+            return repository.add(dto.getName(), dto.toString()) > 0;
+        }
+        return false;
```

Comment:
false? Is it normal behavior when the first point not equal to the last one?
Probably let's move this validation to the object